compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1890,29 +1890,31 @@ object Types {
     }
 
     /** The signature of the last known denotation, or if there is none, the
-     *  signature of the symbol
+     *  signature of the symbol. Signatures are always computed before erasure, since
+     *  some symbols change their signature at erasure.
      */
-    protected def computeSignature(implicit ctx: Context): Signature = {
+    protected def computeSignature(implicit ctx: Context): Signature =
       val lastd = lastDenotation
-      if (lastd != null) lastd.signature
+      val isErased = ctx.erasedTypes
+      if lastd != null && !isErased then lastd.signature
+      else if isErased then computeSignature(using ctx.withPhase(ctx.erasurePhase))
       else symbol.asSeenFrom(prefix).signature
-    }
 
     /** The signature of the current denotation if it is known without forcing.
      *  Otherwise the signature of the current symbol if it is known without forcing.
      *  Otherwise NotAMethod.
      */
     private def currentSignature(implicit ctx: Context): Signature =
```

Comment:
`currentSignature` is passed as an argument to `disambiguate` which passes it to `Denotation#atSignature`. If we're after erasure and the signature of the denotation is different from the pre-erasure signature that `currentSignature`, then `atSignature` would fail to match anything, wouldn't it? Or is it this OK because overloaded denotations do not appear after erasure at all? If so then `currentSignature` wouldn't need to do anything special.

Reply:
Yes, after erasure all references should be symbolic, so there's no need to do overloading resolution.